### ISSUE TITLE
Cannot match stubs using ImplicitlyUnwrappedOptional (!)

### ISSUE THREAD

**Issue by menehune23** [Brightify/Cuckoo#96]
When creating a stub for a method that uses an exclamation point on a parameter, a call to the stubbed method results in:

```
fatal error: No stub for method `foo(i: Int!) -> String` using parameters 5 and no original implementation was provided.
```

It seems that this is because Cuckoo doesn't support `ImplicitlyUnwrappedOptional`.

### Code to reproduce

```swift
protocol FooProtocol {
    func foo(i: Int!) -> String
}

func testFoo() {
    let myFoo: FooProtocol = MockFooProtocol()

    stub(myFoo as! MockFooProtocol) { stub in
        when(stub.foo(i: anyInt())).thenReturn("test")
    }
    
    expect(myFoo.foo(i: 5)) == "test"
}
```

Changing the `!` to a normal optional `?` works.

Using version **0.8.2**

**Comment by Mackarous** [comment]
Also noticing this issue myself

**Comment by TadeasKriz** [comment]
@Mackarous Which Cuckoo version? I just tried it and can't reproduce it. Thanks.

**Comment by TadeasKriz** [comment]
I think I know what's the problem. Try using `equal(to: $)` instead of just `$` ($ is the value, for example `5`). It's an unfortunate limitation of Swift.

**Comment by soupangel** [comment]
Having a similar issue, latest version of Cuckoo, Swift 4. Let's say I have a class:

```
class MyClass{
     var baseURL: String!
}
```

And now in my test class:

```
func testBaseURL(){
  
    let mock = MockMyClass()
    stub(mock){stub in
        when(stub.baseURL.get).thenReturn("www.example.com")
    }

    classForTesting.doSomething(with: mock) //classForTesting uses myClass.baseURL

}
```

results in 

`MockManager.swift:57: error: -[MyTests testBaseURL] : failed - No stub for method 'baseURL#get' using parameters () and no original implementation was provided.'`

if I declare `var baseURL: String?` in MyClass instead, I don't get the same error. So it seems Cuckoo is tripping up over stubbing non-optional properties?

**Comment by TadeasKriz** [comment]
@soupangel Thanks, I was able to reproduce this. Problem is the implicitly unwrapped optional. I'll look into it.

**Comment by ealymbaev** [comment]
Any updates on this issue?

**Comment by sanojbluejeans** [comment]
did you able to solve it? It was working fine as soon I started using Xcode this same issue is coming.

> caught "NSInternalInconsistencyException", "No stub for method `getMediaStates() -> MediaState` using parameters ()."
even though I have the method in place. any help would be appreciated.

**Comment by MatyasKriz** [comment]
Hello everyone, I have tried all the available code for reproducing this issue in `0.12.0` that I'm using as reference for the issues that I'm resolving.

I am not able to reproduce this issue. @menehune23's code works perfectly and with @soupangel's I wasn't sure what to do with the `baseURL` afterwards, so a simple `assert(mock.baseURL == "expected")` worked fine.

I'm closing this issue for now until someone can reliably reproduce it in `0.12.0` or later.
